spec/workflow/states/wait_spec.rb

```diff
@@ -0,0 +1,22 @@
+RSpec.describe Floe::Workflow::States::Pass do
+  let(:workflow) { Floe::Workflow.load(GEM_ROOT.join("examples/workflow.asl")) }
+  let(:state)    { workflow.states_by_name["WaitState"] }
+
+  describe "#run!" do
+    it "sets the result to the result path" do
+      allow_any_instance_of(Object).to receive(:sleep).with(state.seconds)
```

Comment:
@Fryguy do you know a better way to stub `sleep()` ?  I could call it from a sleep instance method on the class but that feels like making an awkward change just for test stubbing

Reply:
I think you can can stub sleep directly on the state, because technically it calls `self#sleep`, _then_ walks the inheritance chain.

```suggestion
      except(state).to receive(:sleep).with(state.seconds)
```

Reply:
Oh yeah that does work, I assumed that would be looking for a `Floe::Wait#sleep` method